task/git-cli/0.3/git-cli.yaml

```diff
@@ -0,0 +1,139 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-cli
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.21.0"
+    tekton.dev/categories: Git
+    tekton.dev/tags: git
+    tekton.dev/displayName: "git cli"
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This task can be used to perform git operations.
+
+    Git command that needs to be run can be passed as a script to
+    the task. This task needs authentication to git in order to push
+    after the git operation.
+
+  workspaces:
+    - name: source
+      description: A workspace that contains the fetched git repository.
+
+    - name: input
+      description: A workspace that contains file that needs to be added to git.
+
+    - name: ssh-directory
+      optional: true
+      description: |
+        A .ssh directory with private key, known_hosts, config, etc. Copied to
+        the user's home before git commands are executed. Used to authenticate
+        with the git remote when performing the clone. Binding a Secret to this
+        Workspace is strongly recommended over other volume types.
+
+    - name: basic-auth
+      optional: true
+      description: |
+        A Workspace containing a .gitconfig and .git-credentials file. These
+        will be copied to the user's home before any git commands are run. Any
```

Comment:
Can we make this as a common workspace and in the script check for the corresponding credentials ? On one side it can be easy for the user to remember that for ssh there is a different workspace and for basic auth the workspace is different but on one hand the workspaces list increases :sweat_smile:

Reply:
I'm not sure if that would make it more maintainable. Both the authorization workspaces and the code are now cleanly separated. The entire ssh workspace even gets copied in its entirety, so I don't know on what files to filter (`id_rsa`, `config` and `known_hosts` are just suggestions). I mean we could probably do it we really want to, but I think it's for little gain.

Reply:
sure! makes sense. Thanks